### ISSUE TITLE
:dojo should retain buffer until request finishes

### ISSUE THREAD

**Issue by ohAitch** [urbit/arvo#63]
I remember at some point a crash would leave your input where it was to let you fix it, instead of accepting the input into the history.

Similarly, a blocked dojo request should show a spinner over the _entered command_, making it obvious why your shell is rejecting input. And perhaps accept `^C` instead of backspace to cancel, @cgyarvin?

**Comment by ohAitch** [comment]
I guess a specific thing this is blocked on is deferred `%nice`.

**Comment by jtobin** [comment]
Closing this one as I think it's a bit too idiosyncratic to be useful more broadly.

**Comment by ohAitch** [comment]
Disagree that this is idiosyncratic, it concerns all asynchronous shell commands.

**Comment by philipcmonk** [comment]
"deferred `%nice`" (nowadays, `%coup`) is still not a thing, though there's been recent interest in reviving it.
